macos.py

```diff
@@ -0,0 +1,308 @@
+import platform
+import os
+import subprocess
+
+
+print("=====================")
+print("| FLUTTER INSTALLER |")
+print("=====================")
+print(" ")
+print("System Specifications: ")
+print("-----------------------")
+
+# System Check Variables
+cpu = platform.processor()
+os = platform.system()
+
+print("CPU: ", cpu)
+print("OS: ", os)
+
+# Check Application Requirements
+print(" ")
+print("Application Requirements: ")
+print("--------------------------")
+
+# Xcode CLI
+try:
+    subprocess.run(["xcode-select", "--version"], check=True)
+    xCLI = ("Installed")
+except subprocess.CalledProcessError:
+    xCLI = ("Not Installed")
+
+print("Xcode Command Line Tools: ", xCLI)
+
+# Homebrew
+
+
+def is_brew_installed():
+    try:
+        subprocess.run(['which', 'brew'], check=True, stdout=subprocess.PIPE)
+        return True
+    except subprocess.CalledProcessError:
+        return False
+
+
+if is_brew_installed():
+    brew = ("Installed.")
+else:
+   brew = ("Not Installed.")
+
+
+print("Homebrew: ", brew)
+
+# Cocoapods
+
+
+def is_cocoapods_installed():
+    try:
+        subprocess.run(['which', 'pod'], check=True, stdout=subprocess.PIPE)
+        return True
+    except subprocess.CalledProcessError:
+        return False
+
+
+if is_cocoapods_installed():
+    cPods = ("Installed")
+else:
+    cPods = ("Not Installed")
+
+print("Cocoapods: ", cPods)
+
+
+# Flutter
+def check_flutter_exists():
+  try:
+    subprocess.run(['which', 'flutter'], check=True,
+                   stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    return True
+  except subprocess.CalledProcessError:
+    return False
+
+
+if check_flutter_exists():
+  flutter = ("Installed")
+else:
+  flutter = ("Not Installed")
+
+print("Flutter: ", flutter)
+
+# Xcode
+
+
+def check_xcode_installed():
+    try:
+        xcode_path = subprocess.check_output(["xcode-select", "--print-path"])
+        return True
+    except subprocess.CalledProcessError:
+        return False
+
+
+if check_xcode_installed():
+    xcode = ("Installed")
+else:
+    xcode = ("Not Installed")
+
+print("Xcode: ", xcode)
+
+# VS Code
+
+
+def is_vscode_installed():
+    try:
+        subprocess.run(["which", "code"], check=True, capture_output=True)
+        return True
+    except subprocess.CalledProcessError:
+        return False
+
+
+if is_vscode_installed():
+    vscode = ("Installed")
+else:
+    vscode = ("Not Installed")
+
+print("VS Code: ", vscode)
+
+
+# INSTALLATION
+print("")
+print("")
+print("INSTALLATION")
+print("----------")
+
+# if Apple Silicon Macs
+if cpu == "arm":
```

Comment:
check if rosetta is installed first. if installed, add a prompt that tells the user if they need to update rosetta or not

Reply:
Noted, going to review further Flutter's Documentation.